ghproxy/ghcache/ghcache.go

```diff
@@ -39,18 +39,32 @@ import (
 	"golang.org/x/sync/semaphore"
 )
 
+type CacheResponseMode string
+
 // Cache response modes describe how ghcache fulfilled a request.
 const (
-	ModeError   = "ERROR"    // internal error handling request
-	ModeNoStore = "NO-STORE" // response not cacheable
-	ModeMiss    = "MISS"     // not in cache, request proxied and response cached.
-	ModeChanged = "CHANGED"  // cache value invalid: resource changed, cache updated
+	CacheModeHeader = "X-Cache-Mode"
+
+	ModeError   CacheResponseMode = "ERROR"    // internal error handling request
+	ModeNoStore CacheResponseMode = "NO-STORE" // response not cacheable
+	ModeMiss    CacheResponseMode = "MISS"     // not in cache, request proxied and response cached.
+	ModeChanged CacheResponseMode = "CHANGED"  // cache value invalid: resource changed, cache updated
 	// The modes below are the happy cases in which the request is fulfilled for
 	// free (no API tokens used).
-	ModeCoalesced   = "COALESCED"   // coalesced request, this is a copied response
-	ModeRevalidated = "REVALIDATED" // cached value revalidated and returned
+	ModeCoalesced   CacheResponseMode = "COALESCED"   // coalesced request, this is a copied response
+	ModeRevalidated CacheResponseMode = "REVALIDATED" // cached value revalidated and returned
 )
 
+func CacheModeIsFree(mode CacheResponseMode) bool {
```

Comment:
Technically also `ERROR` right? Since we never hit GitHub

Reply:
We don't return a response in the error case so a X-Cache-Mode header never has the Error response mode.
I can add it anyways though.